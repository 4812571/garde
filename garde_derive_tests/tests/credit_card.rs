#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(credit_card)]
    field: &'a str,
}

#[test]
fn credit_card_valid() {
    util::check_ok(
        &[
            Test {
                field: "4539571147647251",
            },
            Test {
                field: "343380440754432",
            },
        ],
        &(),
    )
}

#[test]
fn credit_card_invalid() {
    util::check_fail!(
        &[
            Test {
                field: "zduhefljsdfKJKJZHUI"
            },
            Test {
                field: "5236313877109141"
            },
        ],
        &()
    )
}
