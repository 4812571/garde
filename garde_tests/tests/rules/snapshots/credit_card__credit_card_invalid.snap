---
source: garde_derive_tests/tests/credit_card.rs
expression: snapshot
---
Test {
    field: "zduhefljsdfKJKJZHUI",
}
value.field: not a valid credit card number: invalid format

Test {
    field: "5236313877109141",
}
value.field: not a valid credit card number: invalid luhn


